ja/lessons/basics/functions.md

```diff
@@ -110,15 +119,177 @@ iex> Greeter2.hello("Fred", "Jane")
 "Hello, Fred and Jane"
 ```
 
-関数名の一覧を上記のコメントに載せました。例えば、1つめの実装は引数を取らないので`hello/0`、2つ目は1つの引数を取るので`hello/1`となります。他の言語におけるオーバーロードとは違い、これらは互いに _異なる_ 関数として扱われます。(さっき扱ったパターンマッチングは _同じ_ 数の引数を取る関数定義が複数ある場合のみ適用されます)
+関数名の一覧を上記のコメントに載せました。
+例えば、1つめの実装は引数を取らないので`hello/0`、2つ目は1つの引数を取るので`hello/1`となります。
+他の言語におけるオーバーロードとは違い、これらは互いに _異なる_ 関数として扱われます。
+(さっき扱ったパターンマッチングは _同じ_ 数の引数を取る関数定義が複数ある場合のみ適用されます)
+
+### 関数とパターンマッチング
+
+内部では、関数は実行された時の引数をパターンマッチングしています。
+
+マップを受け取るが、特定のキーにだけ関心がある関数が必要であるとしましょう。
+私たちは次のようにキーの有無に基づいて引数をパターンマッチすることができます:
+
+```elixir
+defmodule Greeter1 do
+  def hello(%{name: person_name}) do
+    IO.puts "Hello, " <> person_name
+  end
+end
+```
+
+今度はFredという名前の人物を表すマップを持っているとしましょう。
+```elixir
+iex> fred = %{
+...> name: "Fred",
+...> age: "95",
+...> favorite_color: "Taupe"
+...> }
+```
+
+`Greeter1.hello/1  `を`fred`のマップで実行するとこのような結果となります:
+
+```elixir
+# call with entire map
+...> Greeter1.hello(fred)
+"Hello, Fred"
+```
+
+`:name`キーを含まないマップで関数を実行するとどうなるでしょうか？
+
+```elixir
+# call without the key we need returns an error
+...> Greeter1.hello(%{age: "95", favorite_color: "Taupe"})
+** (FunctionClauseError) no function clause matching in Greeter3.hello/1
```

Comment:
It seems original text is wrong, not, `Greeter3`, but `Greeter1` :eyes:
https://github.com/elixirschool/elixirschool/pull/1668